File: app/src/test/java/nerd/tuxmobil/fahrplan/congress/navigation/RoomForC3NavConverterTest.java
package nerd.tuxmobil.fahrplan.congress.navigation;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.JUnit4;

import static org.assertj.core.api.Assertions.assertThat;

@RunWith(JUnit4.class)
public class RoomForC3NavConverterTest {

    @Test
    public void convertWithAdams() {
        assertThat(RoomForC3NavConverter.convert("Adams")).isEqualTo("hall-a");
    }

    @Test
    public void convertWithBorg() {
        assertThat(RoomForC3NavConverter.convert("Borg")).isEqualTo("hall-b");
    }

    @Test
    public void convertWithClarke() {
        assertThat(RoomForC3NavConverter.convert("Clarke")).isEqualTo("hall-c");
    }

    @Test
    public void convertWithDijkstra() {
        assertThat(RoomForC3NavConverter.convert("Dijkstra")).isEqualTo("hall-d");
    }

    @Test
    public void convertWithEliza() {
        assertThat(RoomForC3NavConverter.convert("Eliza")).isEqualTo("hall-e");
    }

    @Test
    public void convertWithNonExisting() {
        assertThat(RoomForC3NavConverter.convert("NonExisting")).isEmpty();
    }

    @Test
    public void convertWithNull() {
        assertThat(RoomForC3NavConverter.convert(null)).isEmpty();
    }

}

File: app/src/test/java/nerd/tuxmobil/fahrplan/congress/navigation/RoomForC3NavConverterTest.kt
package nerd.tuxmobil.fahrplan.congress.navigation

import com.google.common.truth.Truth.assertThat
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.MethodSource

class RoomForC3NavConverterTest {

    companion object {

        private fun scenarioOf(roomName: String?, expectedText: String) =
                arrayOf(roomName, expectedText)

        @JvmStatic
        fun data() = listOf(
                scenarioOf(roomName = "Ada", expectedText = "hall-a"),
                scenarioOf(roomName = "Borg", expectedText = "hall-b"),
                scenarioOf(roomName = "Clarke", expectedText = "hall-c"),
                scenarioOf(roomName = "Dijkstra", expectedText = "hall-d"),
                scenarioOf(roomName = "Eliza", expectedText = "hall-e"),
                scenarioOf(roomName = "NonExisting", expectedText = ""),
                scenarioOf(roomName = "", expectedText = ""),
                scenarioOf(roomName = null, expectedText = ""),
        )
    }

    @ParameterizedTest(name = """{index}: room = "{0}" -> expectedText = "{1}"""")
    @MethodSource("data")
    fun convert(
        roomName: String?,
        expectedText: String,
    ) {
        assertThat(RoomForC3NavConverter().convert(roomName)).isEqualTo(expectedText)
    }

}
